### DISCUSSION TITLE
Client Side Caching implementation

### DISCUSSION THREAD

**Discussion by wojciechbator** [dahomey-technologies/rustis#55] [Ideas]
Hello, I'm migrating services from golang to rust and it seems that there's not a single library in rust ecosystem that address Redis feature of client side caching: https://redis.io/docs/manual/client-side-caching/ 

An example implementation of this is in rueidis library for golang - https://github.com/redis/rueidis/blob/main/cache.go which is very useful if one has a subset of known keys that he/she would like to reuse for a given time period. Would you consider supporting this also in rustis? [UPD]: Seems that redis-rs authors are also planning this - https://github.com/redis-rs/redis-rs/issues/826.

Regards

**Comment by mcatanzariti** [comment]
Hello @wojciechbator,

What would be your use case with client side caching? opt-in? opt-out? broadcasting?

IMHO, not all the read commands could be cached. We could cache results from GET, MGET, eventually HGET/HGETALL/SMEMBERS but no LIST or SORTED SET commands. Any opinion on this topic?

Cheers,

Michaël

**Comment by mcatanzariti** [comment]
Hello @wojciechbator,

From release [0.16.0](https://github.com/dahomey-technologies/rustis/releases/tag/0.16.0), client-side caching is supported.

Cheers,

Michaël
